### PR TITLE
extract factory gen

### PR DESCRIPTION
**Pull request by mitchdraft**

### CODE REVIEW

File: projects/gloo/pkg/syncer/setup_syncer.go [hunk 1]
```diff
@@ -190,13 +137,18 @@ func (s *setupSyncer) Setup(ctx context.Context, kubeCache kube.SharedCache, mem
...
+		return err
+	}
 	opts := bootstrap.Opts{
 		WriteNamespace:  writeNamespace,
 		WatchNamespaces: watchNamespaces,
-		Upstreams:       upstreamFactory,
-		Proxies:         proxyFactory,
-		Secrets:         secretFactory,
-		Artifacts:       artifactFactory,
+		Upstreams:       bf.Upstreams,
```
Comment (rickducott) [inline projects/gloo/pkg/syncer/setup_syncer.go]: This isn't strictly necessary, we could alternatively just add other params to bf and return that -- reduces the need to update 2 places (here and the `BootstrapFactories` function) when we add a new factory

Comment (mitchdraft) [reply projects/gloo/pkg/syncer/setup_syncer.go]: I was torn but yeah, I agree that is better, updated

**Pull request merged**